Fix EOL in C# project files (#109)

Yury-Fridlyand opened the pull request:
1. Minor editor config update
2. Whitespace changes:

`\r\n` changed to `\n` by
```
find . -type f | xargs dos2unix
```

shachlanAmazon commented:
please no style changes without a linter rule to enforce them.

shachlanAmazon commented:
What's the .editorconfig change? I can't see it, because of the linebreak change.

shachlanAmazon commented:
if you change the global editorconfig, you'll need to trigger all of the linters, and check whether this affects other languages. At least the prettier linter takes this into account.

Yury-Fridlyand commented:
I reworked editorconfig - I moved C# related block (98% of the file) to `csharp` dir and created a symlink into `benchmark/csharp` to avoid duplicating a big file.
I tried to trigger all our CI in 3ece3b0 and they pass.

IMO no linters we have use editorconfig - it is an IDE recommendation ruleset for new lines, new files, new names, etc, not a requirement.
Prettier uses `.editorconfig` as a config by default, but it is not enabled on our project. It doesn't lint `csproj` files though - I tried.

shachlanAmazon commented:
@avifenesh rebase over you CI change this and verify that it doesn't change the behavior of prettier.

Sa1Gur commented:
BTW we could make add dotnet_analyzer_diagnostic.category-Style.severity = error to .editorconfig and it will make all codestyle rule breach as error. Not sure if it is desired outcome

shachlanAmazon commented:
@Sa1Gur good idea. I don't see the value in warnings - they're just unenforced noise. Either we want the CI to enforce the linting, or we don't care about it.

Sa1Gur commented:
Ok. I will raise PR then

barshaul commented:
@Yury-Fridlyand can it be merged?
